Document and make public new issue lifecycle (#3465)
nginxinc/kubernetes-ingress: CONTRIBUTING.md

## Ask a Question

> To ask the core development team a question please use the [Github Discussions](https://github.com/nginxinc/kubernetes-ingress/discussions) feature.

Reviewer: Consider simplifying to "please use GitHub Discussions"

> To ask the core development team a question please use the [Github Discussions](https://github.com/nginxinc/kubernetes-ingress/discussions) feature.

Reviewer 2: ```suggestion
To ask a question please use the [Github Discussions](https://github.com/nginxinc/kubernetes-ingress/discussions) feature.
```
I think the question is for anybody that know the answer 🙂

You can also join our [Community Slack channel](https://community.nginx.org/joinslack) which has a wider NGINX audience.

> Please reserve Github issues for specific feature requests and bugs rather than general questions.

Reviewer 3: ```suggestion
Please reserve Github issues for feature requests and bugs rather than general questions.
```
No need to have the word 'specific', because the nature of feature requests means they're more than likely going to be inherently specific anyway.

### Report a Bug

To report a bug, open an issue on GitHub and choose the type 'Bug report'. Please ensure the issue has not already been reported, and that you fill in the template as provided, as this can reduce turnaround time.

### Suggest a new feature or other improvement

To suggest an new feature or other improvement, create an issue on Github and choose the type 'Feature request'. Please fill in the template as provided.

### Open a Pull Request

> * Before working on a possible pull request, you should first open an associated issue (feature request or bug report) describing the proposed change. This gives an opportunity for the core development team to discuss the potential pull request with you before you do the work.

Reviewer 4: ```suggestion
* Before working on a possible pull request, first open an associated issue describing the proposed change. This allows the core development team to discuss the potential pull request with you before you do the work.
```

> * Before working on a possible pull request, you should first open an associated issue (feature request or bug report) describing the proposed change. This gives an opportunity for the core development team to discuss the potential pull request with you before you do the work.

Reviewer 2: I don't think we should require an issue for every PR. It might be needed for some big or controversial change, but in most cases seeing the code changes actually helps. We probably should change the PR template to encourage people to give a more detailed description of the PR.

Author: I can see the value in that. However as we want to move towards a 'github first' workflow in general, I think all work by external contributors should start with an issue. This will allow us to apply the issue workflow (and perhaps kindly turn down the issue) before the contributor goes to the trouble of creating a PR, thus saving them work.

If we decide that it is too onerous, we can always implement the changed PR template at that point.

Reviewer 2: we can apply the issue workflow to PRs as well

* Fork the repo, create a branch, submit a PR when your changes are tested and ready for review
* Fill in [our pull request template](https://github.com/nginxinc/kubernetes-ingress/blob/main/.github/PULL_REQUEST_TEMPLATE.md)

> **Note**: Don't forget to create a feature request / bug report issue first to start a discussion about the proposed change.

Reviewer 4: ```suggestion
**Note**: Remember to create a feature request / bug report issue first to start a discussion about the proposed change.
```
